app/models/manageiq/providers/embedded_automation_manager/configuration_script_source.rb

```diff
@@ -85,11 +85,22 @@ def sync
     raise NotImplementedError, N_("sync must be implemented in a subclass")
   end
 
-  def checkout_git_repository(target_directory)
+  def checkout_git_repository(target_directory = nil)
     return if git_repository.nil?
 
+    target_directory_given = !!target_directory
+    target_directory     ||= Dir.mktmpdir
+
     git_repository.update_repo
     git_repository.checkout(scm_branch, target_directory)
+
+    return target_directory unless block_given?
+
+    begin
+      yield Pathname.new(target_directory)
```

Comment:
Minor inconsistency here, but if a block is given we yield a Pathname whereas with no block given we return the target_directory, which is a String (if Dir.mktmpdir) or whatever is passed in (likely a String, but could also be a Pathname).  I'm thinking these types should be made consistent.